src/server/project.js

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+var fs = require('fs'),
+    path = require('path'),
+    cordovaServe = require('cordova-serve'),
+    Q = require('q'),
+    dirs = require('./dirs'),
+    pluginUtil = require('./utils/plugins'),
+    prepareUtil = require('./utils/prepare'),
+    utils = require('./utils/jsUtils');
+
+/**
+ * @constructor
+ */
+function Project(simulator, platform) {
+    this._simulator = simulator;
```

Comment:
That sounds good, I will change it to `_simulatorProxy` :)
Doing what you suggest is a good option, but there are some points that I consider when using this proxy object that acts as a kind of interface:
- telemetry and config are widely used across all of the other instances, and big candidates to still be widely used if we are new models/objects to the server
- if I pass through parameter both of them, on each instance that uses it, I'm duplicating information in all over the places, when actually the config and telemetry are not likely to be properties of each single object, but to the simulator itself, who coordinates the interaction between all of the components
- The simulatorProxy acts a middle instance that provides common functionality across all of the objects, and I guess that this approach is more flexible and will provide us a clean way to even add more functionality. Today we need what we have right now, but we can easily expose status APIs, so other instances checks the status before performing critical actions that modifies or generates files as a result of asynchronous operations, for example (thinking out loud)